q2_metadata/normalization/_norm_check_rules.py

```diff
@@ -0,0 +1,311 @@
+    possible_validations = {
+        'force_to_blank_if': {'is null'}
+    }
+    # check the first level is a dict
+    if isinstance(rule_value, dict):
+        # get the current keys
+        rule_value_set = set(rule_value)
+        # get the accepted keys from the current keys
+        accepted_set = rule_value_set & set(possible_validations)
+        # if there is not more keys than the accepted ones
+        if not len(rule_value_set ^ accepted_set):
+            # get the possible conditions
+            possible_validations_2 = possible_validations['force_to_blank_if']
+            # for each of the condition/replacements for the accepted keys
+            for rule_2, rule_value_2 in rule_value['force_to_blank_if'].items():
+                # check that the content of replacements are lists
+                if isinstance(rule_value_2, list):
+                    # get the accepted conditions from the current conditions
+                    if rule_2 in possible_validations_2:
+                        return True
+    return False
+
+
+def check_normalization(rule_value):
+    """Check that the user-defined value for the
+    rule "normalization" is correctly formatted.
+
+    normalization must be a dict
+        Apply the rules of normalizing the variables values, by replacing
+        the values outside the range by a given replacement value.
+                    normalization:
+                        gated_value:
+                            - Out of bounds
+                        maximum:
+                            - int
+                            - float
+                        minimum:
+                            - int
+                            - float
+
+    Parameters
+    ----------
+    rule_value
+        Parsed rule value for the rule "normalization".
+
+    Returns
+    -------
+    bool
+        True if successful, False otherwise.
+    """
+    if isinstance(rule_value, dict):
+        possible_normalizations = {'maximum', 'minimum', 'gated_value'}
+        rule_value_set = set(rule_value)
+        if not len(rule_value_set ^ (rule_value_set & possible_normalizations)):
+            return True
+    return False
+
+
+def check_format(rule_value):
+    """Check that the user-defined value for
+    rule "format" is correctly formatted.
+
+    format must be a str
+        Apply rule of checking that the passed column in the correct format:
+        [THIS COMMAND WOULD EITHER INFER THE DTYPE (as of now) OR USE THE
+        RESULT OF AN INITIAL DTYPE GETTER FUNCTION (see _dtypes.py)]
+                    format:
+                      - bool
+                      - float
+                      - int
+                      - str
+
+    Parameters
+    ----------
+    rule_value
+        Parsed rule value for rule "format".
+
+    Returns
+    -------
+    bool
+        True if successful, False otherwise.
+    """
+    allowed_values = {'bool', 'float', 'int', 'str'}
```

Comment:
why are types represented as strings?

Reply:
in the code that will apply the rules, these will be checked. For example, one may want to make sure a variable is series of integers: we have this rule, and the code will be something liek:
```
if variable.rules.format=='str' and metadata.variable.dtype!=str:
    # problem to handle
```
The `format` as a rule was discussed amongst us (Rebecca and I) and we thought about having it hard coded somewhere else so that a user cannot change what a variable is supposed to be made of: that might be true for AGP (e.g. variable `bmi` could be hard-coded `format: int`), but it's not that easy for lambda users (e.g. may want to have variable `size_of_the_tree` set to format `str`...)